osgp/protocol-adapter-dlms/osgp-protocol-simulator-dlms/simulator/dlms-device-simulator/src/main/java/org/opensmartgridplatform/simulator/protocol/dlms/cosem/EMonthlyBillingValuesPeriod1DSMR4.java

```diff
@@ -22,7 +24,7 @@
 import org.opensmartgridplatform.simulator.protocol.dlms.cosem.processing.UInteger32DataProcessor;
 
 @CosemClass(id = 7)
-public class DataOfBillingPeriod1 extends ProfileGeneric {
+public class EMonthlyBillingValuesPeriod1DSMR4 extends ProfileGeneric {
```

Comment:
Although I like the use of the more descriptive name with `MonthlyBillingValues`, I'm not sure if adding classes for specific profiles (e.g. DSMR4 and SMR5) is the way to go. When GXF will be used for more profiles, we will get many more classes. My suggestion would be to pass more information from the profiles (e.g. the obis code, the list with capture objects etc) to the constructor of this class, so it can be more generic.

Reply:
Might be a good idea, but I'll suggest this will be done in another story were DSMR4 could be more isolated.
I'm not sure if this will make things easier to understand/maintain, but it will remove some code duplication that can not be removed due to way the jDLMS library works (reflection?).